base-infrastructure/terraform/app_resources/database.tf

```diff
@@ -0,0 +1,12 @@
+resource "azurerm_postgresql_flexible_server_database" "app" {
+  count = var.database_config.create_database ? 1 : 0
+
+  name      = var.database_config.database_name
+  server_id = var.database_config.server_id
+  collation = "en_US.utf8"
+  charset   = "utf8"
+
+  lifecycle {
+    prevent_destroy = false
```

Comment:
why `false` as default? Should we make it `true` by default and allow overriding by a variable if required?

Reply:
Thanks for spotting that @sunu , I actually intended to have it as true. I've fixed that now.